python/lib/dependabot/python/update_checker/index_finder.rb

```diff
@@ -165,9 +165,10 @@ def clean_check_and_remove_environment_variables(url)
             map { |u| u.strip.gsub(%r{/*$}, "") + "/" }
 
           regexp = url.
-                   sub(%r{(?<=://).+@}, "").
+                   sub(%r{(?<=://).+@}, "@").
                    split(ENVIRONMENT_VARIABLE_REGEX).
                    map { |part| Regexp.quote(part) }.
+                   map { |part| part.sub("@", ".+") }.
                    join(".+")
           authed_url = config_variable_urls.find { |u| u.match?(regexp) }
```

Comment:
I tried to apply this patch but it doesn't help me with an issue is described here: #1772 
I added some debug prints and found that config_variable_urls is empty.

In my case I put the extra-index-url in the requirements.txt:
```pip
--extra-index-url https://${PYPI_USER}:${PYPI_PASSWORD}@<....>/artifactory/api/pypi/<....>/simple/
```
this patch gives a correct regexp, but it doesn't help because of empty config_variable_urls
@feelepxyz @orf Any thoughts or hints?

Reply:
Look like I found, I just added the index_url to credentials dict in dependabot-script. But I really thought it should be parsed automatically somehow

Reply:
@SVilgelm does the change i've added is needed to use this fix?
Can you tell me what do you had to change for it to work ok?
Thanks